### PR TITLE
Run flatc only when fusion_cache.fbs is modified

### PR DESCRIPTION
**Pull request by jacobhinkle**
I noticed that cmake is running flatc for every build lately. I think this PR fixes it by explicitly stating that `fusion_cache_generated.h` depends on `fusion_cache.fbs`. I got this pattern from https://samthursfield.wordpress.com/2015/11/21/cmake-dependencies-between-targets-and-files-and-custom-commands/.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -218,10 +218,18 @@ if (PROJECT_IS_TOP_LEVEL)
...
-      COMMAND ./third_party/flatbuffers/flatc -o ../csrc/serde/ -c -b ../csrc/serde/fusion_cache.fbs
-      DEPENDS flatc
-      COMMENT "Generating fusion_cache_generated header from fusion_cache.fbs")
+  add_custom_command(
+    OUTPUT
+      ${NVFUSER_ROOT}/csrc/serde/fusion_cache_generated.h
+    DEPENDS
+      ${NVFUSER_ROOT}/csrc/serde/fusion_cache.fbs
+    DEPENDS flatc
+    COMMAND ./third_party/flatbuffers/flatc -o ../csrc/serde/ -c -b ../csrc/serde/fusion_cache.fbs
```
Comment (rdspring1) [inline CMakeLists.txt:227]: Should we use `${NVFUSER_ROOT}` here?

Comment (jacobhinkle) [reply CMakeLists.txt:227]: Not sure what the best way is to refer to that path. @jjsjann123 do you have any thoughts?

**Review by rdspring1 — APPROVED** [review]
LGTM. 

I also thought about running `lintrunner -a ./csrc/serde/fusion_cache_generated.h`

File: CMakeLists.txt (hunk 1 above)
Comment (jacobhinkle) [reply CMakeLists.txt]: Oh I see what you mean. Just for the path arguments. Pushed a change that does that and refers to flatc binary location inside `${CMAKE_CURRENT_BINARY_DIR}`.

**Pull request merged**